src/gennylib/src/PoolFactory.cpp

```diff
@@ -192,28 +194,59 @@ std::string PoolFactory::makeUri() const {
 }
 
 mongocxx::options::pool PoolFactory::makeOptions() const {
-    mongocxx::options::tls tlsOptions;
+    mongocxx::options::client clientOptions;
```

Comment:
I'll revert the `std::move` so it remains true to the original. 

As for setting the `apm_opts`, the original code was creating a copy of the entire `options::client` object because it cannot just simply add the new options to the existing `options::client` within `options::pool` as it is `const`ed & read only. Since I've unified all the options setting in this function, I can just set the options directly in the `clientOptions`, before constructing the `options::pool`.